### PR TITLE
Reset password flow

### PR DESCRIPTION
**Pull request by wariofdz**
### Added:

- Added new function `getUpdateUserMeErrorMessage.ts` and his test to errors from API.
- Added new file `updateUserMeErrorMessage.ts` with error messages.
- Added new structure for data and actions to the new component and hook for reset password flow: `UseResetPasswordActions.ts, UseResetPasswordData.ts, UseResetPasswordStatus.ts.`
- Added new hook `useResetPassword.ts` and his test.
- Added new component `ResetPassword.tsx` and his test.
- Added new style for Reset Password page.

### Changed:

- Modified component `AuthRoutes.tsx` to include the new route for reset password.
- Modified function `getSlug.ts` to include the new component.
- Modified enum `PageName.ts` to include the new component.
- Modified style `_index.scss` to include the new component.

### CODE REVIEW

File: packages/frontend/web-ui/src/auth/hooks/useResetPassword.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+            ...resetPasswordData.form.fields,
+          },
+          validation: formValidation,
+        },
+        status: UseResetPasswordStatus.pending,
+      });
+    }
+  }
+
+  function buildErrorMessage(
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/auth/hooks/useResetPassword.ts:207]: We should split this in two functions, one for each endpoint involved

File: packages/frontend/web-ui/src/common/helpers/getSlug.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ const NAVBAR_PAGE_NAME_TO_PAGE_SLUG_MAP: { [TPage in PageName]: string } = {
   [PageName.register]: '/auth/register',
   [PageName.userMe]: '/users/me',
   [PageName.forgot]: '/auth/forgot',
+  [PageName.reset]: '/auth/reset-password',
 };
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/helpers/getSlug.ts:14]: A test case should be added

**Review by notaphplover — CHANGES_REQUESTED** [review]
Nice, I added some suggestions at first glance. I'll review the PR in depth tomorrow :)

File: packages/frontend/web-ui/src/auth/hooks/useResetPassword.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/auth/hooks/useResetPassword.ts:207]: Done!

File: packages/frontend/web-ui/src/common/helpers/getSlug.ts (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/helpers/getSlug.ts:14]: Added!

**Review by notaphplover — APPROVED** [review]

**Pull request merged**